reaktive/src/commonTest/kotlin/com/badoo/reaktive/disposable/DisposableBuilderWithCallbackTest.kt

```diff
@@ -0,0 +1,46 @@
+package com.badoo.reaktive.disposable
+
+import kotlin.test.Test
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class DisposableBuilderWithCallbackTest {
+
+    @Test
+    fun isDisposed_returns_false_after_creation() {
+        val disposable = Disposable {}
+
+        assertFalse(disposable.isDisposed)
+    }
+
+    @Test
+    fun isDisposed_returns_true_IF_disposed() {
+        val disposable = Disposable {}
+
+        disposable.dispose()
+
+        assertTrue(disposable.isDisposed)
+    }
+
+    @Test
+    fun calls_callback_WHEN_disposed() {
+        var isCalled = false
```

Comment:
We use `AtomicBoolean()` in tests for such cases, check `DoOnBeforeErrorTest` as example.

Reply:
Because of freezing by our `TestObserver`, `TestSource`, etc.. There is no freezing here at the moment. Why do you want to use `AtomicBoolean`?